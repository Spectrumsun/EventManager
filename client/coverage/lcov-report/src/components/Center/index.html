<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Center</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> src/components/Center
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">12.23% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>17/139</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/50</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/51</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">13.08% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>17/130</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="AddCenter.js"><a href="AddCenter.js.html">AddCenter.js</a></td>
	<td data-value="4.44" class="pic low"><div class="chart"><div class="cover-fill" style="width: 4%;"></div><div class="cover-empty" style="width:96%;"></div></div></td>
	<td data-value="4.44" class="pct low">4.44%</td>
	<td data-value="45" class="abs low">2/45</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="14" class="abs low">0/14</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="12" class="abs low">0/12</td>
	<td data-value="4.55" class="pct low">4.55%</td>
	<td data-value="44" class="abs low">2/44</td>
	</tr>

<tr>
	<td class="file low" data-value="CenterInfo.js"><a href="CenterInfo.js.html">CenterInfo.js</a></td>
	<td data-value="11.76" class="pic low"><div class="chart"><div class="cover-fill" style="width: 11%;"></div><div class="cover-empty" style="width:89%;"></div></div></td>
	<td data-value="11.76" class="pct low">11.76%</td>
	<td data-value="17" class="abs low">2/17</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="14" class="abs low">0/14</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="9" class="abs low">0/9</td>
	<td data-value="13.33" class="pct low">13.33%</td>
	<td data-value="15" class="abs low">2/15</td>
	</tr>

<tr>
	<td class="file low" data-value="EditCenter.js"><a href="EditCenter.js.html">EditCenter.js</a></td>
	<td data-value="8.33" class="pic low"><div class="chart"><div class="cover-fill" style="width: 8%;"></div><div class="cover-empty" style="width:92%;"></div></div></td>
	<td data-value="8.33" class="pct low">8.33%</td>
	<td data-value="48" class="abs low">4/48</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="18" class="abs low">0/18</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="13" class="abs low">0/13</td>
	<td data-value="8.7" class="pct low">8.7%</td>
	<td data-value="46" class="abs low">4/46</td>
	</tr>

<tr>
	<td class="file low" data-value="PickCenter.js"><a href="PickCenter.js.html">PickCenter.js</a></td>
	<td data-value="27.27" class="pic low"><div class="chart"><div class="cover-fill" style="width: 27%;"></div><div class="cover-empty" style="width:73%;"></div></div></td>
	<td data-value="27.27" class="pct low">27.27%</td>
	<td data-value="11" class="abs low">3/11</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="7" class="abs low">0/7</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="9" class="abs low">3/9</td>
	</tr>

<tr>
	<td class="file low" data-value="ViewCenters.js"><a href="ViewCenters.js.html">ViewCenters.js</a></td>
	<td data-value="27.27" class="pic low"><div class="chart"><div class="cover-fill" style="width: 27%;"></div><div class="cover-empty" style="width:73%;"></div></div></td>
	<td data-value="27.27" class="pct low">27.27%</td>
	<td data-value="11" class="abs low">3/11</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="4" class="abs low">0/4</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="6" class="abs low">0/6</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="9" class="abs low">3/9</td>
	</tr>

<tr>
	<td class="file medium" data-value="getCenter.js"><a href="getCenter.js.html">getCenter.js</a></td>
	<td data-value="66.67" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 66%;"></div><div class="cover-empty" style="width:34%;"></div></div></td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="3" class="abs medium">2/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="1" class="abs low">0/1</td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="3" class="abs medium">2/3</td>
	</tr>

<tr>
	<td class="file low" data-value="pickCenterProp.js"><a href="pickCenterProp.js.html">pickCenterProp.js</a></td>
	<td data-value="25" class="pic low"><div class="chart"><div class="cover-fill" style="width: 25%;"></div><div class="cover-empty" style="width:75%;"></div></div></td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="4" class="abs low">1/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="3" class="abs low">0/3</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="4" class="abs low">1/4</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Apr 15 2018 00:07:57 GMT+0100 (WAT)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
<script src="../../../block-navigation.js"></script>
</body>
</html>
