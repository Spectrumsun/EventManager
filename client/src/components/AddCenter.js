import React from 'react';
import CenterFrom from './CenterForm';

const addCenter = () => (
  <div className="container">
    <div className="card card w-50 loginCard ">
      <div className="card-header dark">
        <h1 className="color">Add Center</h1>
      </div>
      {CenterFrom}
    </div>
  </div>
);

export default addCenter;
