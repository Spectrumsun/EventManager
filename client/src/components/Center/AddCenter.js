/* eslint-disable react/no-unused-state */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import toast from 'toastr';
import axios from 'axios';
import PropTypes from 'prop-types';
import CenterFrom from './Form/CenterForm';
import * as action from '../../store/actions/index';

/**
 * @class AddCenter
 *
 * @extends {React.Component}
 */
class AddCenter extends Component {
    state = {
      name: '',
      city: '',
      address: '',
      about: '',
      availability: '',
      values: '',
      facility: [],
      image: '',
      preview: '',
      imageurl: '',
      publicUrlId: '',
      progress: `${0}%`
    }

  /**
   * @description update facility state
   *
   * @param {any} event
   *
   * @memberof AddCenter
   *
   * @returns {void}
   */
  onClick = () => {
    this.setState({
      facility: this.state.facility.concat([this.state.values])
    });
    this.setState({ values: '' });
  }

  /**
   * @description update component state with current value in dom
   *
   * @param {any} event
   *
   * @memberof AddCenter
   *
   * @returns {void}
   */
   onChange = (e) => {
     this.setState({ [e.target.name]: e.target.value });
   }

  /**
   * @description vaildate data in state and upload the image
   * sends state to api with action dispatch
   * @param {any} event
   *
   * @memberof AddCenter
   *
   * @returns {void}
   */
   onSubmit = (e) => {
     e.preventDefault();
     if (this.state.name === '') {
       toast.error('Center Name cannot be blank');
     } else if (this.state.date === '') {
       toast.error('Center city cannot be blank');
     } else if (this.state.time === '') {
       toast.error('Center Address cannot be blank');
     } else if (this.state.image === '') {
       toast.error('Add an image');
     } else if (this.state.about === '') {
       toast.error('Add about');
     } else if (this.state.purpose === '') {
       toast.error('Center Availability must be set');
     } else if (this.state.facility.length < 1) {
       toast.error('Center facility must be set');
     } else {
       const fd = new FormData();
       const id = `${Date.now()}-${this.state.image.name}`;
       fd.append('file', this.state.image);
       fd.append('public_id', id);
       fd.append('upload_preset', 'eventmanager');
       console.log(fd)
       axios
         .post(
           'https://api.cloudinary.com/v1_1/skybound/image/upload',
           fd, {
             onUploadProgress: (progressEvent) => {
               const level = `${Math.round(progressEvent.loaded /
                progressEvent.total * 100)}%`;
               this.setState({ progress: level });
             }
           }
         )
         .then((response) => {
           this.setState({
             imageurl: response.data.secure_url,
             publicUrlId: response.data.public_id,
             image: null,
             preview: null
           });
           this.props.initPostCenters(
             this.state,
             this.props.history
           );
         })
         .catch((err) => {
           toast.error('Unable to upload. Check your internet');
         });
     }
   }

   /**
   * @description prefent the enter key from submitting the form
   *
   * @param {any} event
   *
   * @memberof AddCenter
   *
   * @returns {void}
   */
   onKeyPress = (e) => {
     if (e.target.type !== 'textarea' && e.which === 13 /* Enter */) {
       e.preventDefault();
     }
   }

  /**
   * @description upload image and return url
   * sends set state to image url
   * @param {any} event
   *
   * @memberof AddCenter
   *
   * @returns {void}
   */
    handleImageChange = (e) => {
      e.preventDefault();
      const reader = new FileReader();
      const file = e.target.files[0];
      reader.onloadend = () => {
        this.setState({
          image: file,
          preview: reader.result,
        });
      };
      reader.readAsDataURL(file);
    }


  /**
   * @description remove add Facility
   *
   * @param {any} event
   *
   * @memberof AddCenter
   *
   * @returns {void}
   */
    removeFacility = (i) => {
      const array = this.state.facility;
      array.splice(i, 1);
      this.setState({ facility: array });
    }

    /**
   * @description renders component to the DOM
   *
   * @memberof AddCenter
   *
   * @returns {JSX} JSX representation of component
   */
    render() {
      const { preview } = this.state;
      let imagePreview = null;
      if (preview) {
        imagePreview = (<img
          src={preview}
          alt="ImagePreview"
          className="imgPre"
        />);
      }

      return (
        <div>
          <div className="container" style={{ paddingTop: '100px' }}>
            <div className="card card w-50 loginCard ">
              <div className="card-header dark">
                <h1 className="color">Add Center</h1>
              </div>
              <CenterFrom
                onChange={this.onChange}
                onSubmit={this.onSubmit}
                centerName={this.state.name}
                city={this.state.city}
                address={this.state.address}
                availability={this.state.availability}
                values={this.state.values}
                handleImageChange={this.handleImageChange}
                onClick={this.onClick}
                removeFacility={this.removeFacility}
                facility={this.state.facility}
                imagePreview={imagePreview}
                about={this.state.about}
                disabled={this.state.values}
                progress={this.state.progress}
                onKeyPress={this.onKeyPress}
              />
            </div>
          </div>
        </div>
      );
    }
}

AddCenter.propTypes = {
  initPostCenters: PropTypes.func.isRequired,
  history: PropTypes.shape({}).isRequired,
};

const mapDispatchToProps = dispatch => ({
  initPostCenters: (input, history) =>
    dispatch(action.initPostCenters(input, history)),
});

export default connect(null, mapDispatchToProps)(AddCenter);
