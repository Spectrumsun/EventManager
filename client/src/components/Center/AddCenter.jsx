/* eslint-disable react/no-unused-state */
import React, { Component } from 'react';
import { connect } from 'react-redux';
import toast from 'toastr';
import PropTypes from 'prop-types';
import CenterFrom from './Form/CenterForm';
import * as action from '../../store/actions/index';
import uploadImage from './ImageUpload';
/**
 * @class AddCenter
 *
 * @extends {React.Component}
 */
export class AddCenter extends Component {
    state = {
      name: '',
      city: '',
      address: '',
      about: '',
      availability: '',
      values: '',
      facility: [],
      image: '',
      preview: '',
      imageurl: '',
      publicUrlId: '',
      progress: `${0}%`,
      formValid: false,
      errorMessage: ''
    }

  /**
   * @description update facility state
   *
   * @param {any} event
   *
   * @memberof AddCenter
   *
   * @returns {void}
   */
  onClick = () => {
    this.setState({
      facility: this.state.facility.concat([this.state.values])
    });
    this.setState({ values: '' });
  }

  /**
   * @description update component state with current value in dom
   *
   * @param {any} event
   *
   * @memberof AddCenter
   *
   * @returns {void}
   */
   onChange = (event) => {
     this.setState({ [event.target.name]: event.target.value });
   }

  /**
   * @description vaildate data in state and upload the image
   * sends state to api with action dispatch
   * @param {any} event
   *
   * @memberof AddCenter
   *
   * @returns {void}
   */
   onSubmit = (event) => {
     event.preventDefault();
     this.setState({ formValid: false });
     if (this.state.name === '') {
       this.setState({ errorMessage: 'Center Name cannot be blank' });
     } else if (this.state.city === '') {
       this.setState({ errorMessage: 'Center city cannot be blank' });
     } else if (this.state.address === '') {
       this.setState({ errorMessage: 'Center address cannot be blank' });
     } else if (this.state.about === '') {
       this.setState({ errorMessage: 'Center about cannot be blank' });
     } else if (this.state.image === '') {
       this.setState({ errorMessage: 'Add an image' });
     } else if (this.state.about === '') {
       this.setState({ errorMessage: 'AAdd about' });
     } else if (this.state.purpose === '') {
       this.setState({ errorMessage: 'AAdd purpose' });
     } else if (this.state.Availability === '') {
       this.setState({ errorMessage: 'Center Availability must be set' });
     } else if (this.state.facility.length < 1) {
       this.setState({ errorMessage: 'Center facility must be set' });
     } else {
       const fd = new FormData();
       const id = `${Date.now()}-${this.state.image.name}`;
       this.setState({ formValid: true });
       fd.append('file', this.state.image);
       fd.append('public_id', id);
       fd.append('upload_preset', 'eventmanager');
       const fileProgress = {
         onUploadProgress: (progressEvent) => {
           const progressMeter = `${Math.round(progressEvent.loaded / progressEvent.total * 100)}%`;
           this.setState({ progress: progressMeter });
           this.setState({ errorMessage: ' ' });
         }
       };
       uploadImage(fd, fileProgress)
         .then((res) => {
           this.setState({
             imageurl: res.data.secure_url,
             publicUrlId: res.data.public_id,
             image: null,
             preview: null,
             formValid: true
           });
           this.props.initPostCenters(
             this.state,
             this.props.history
           );
         })
         .catch(() => {
           toast.error('Unable to upload. Check your internet');
           this.setState({ formValid: false });
           this.setState({ errorMessage: ' ' });
         });
     }
   }

   /**
   * @description prefent the enter key from submitting the form
   *
   * @param {any} event
   *
   * @memberof AddCenter
   *
   * @returns {void}
   */
   onKeyPress = (event) => {
     if (event.target.type !== 'textarea' && event.which === 13 /* Enter */) {
       event.preventDefault();
     }
   }

  /**
   * @description upload image and return url
   * sends set state to image url
   * @param {any} event
   *
   * @memberof AddCenter
   *
   * @returns {void}
   */
    handleImageChange = (event) => {
      event.preventDefault();
      const reader = new FileReader();
      const file = event.target.files[0];
      reader.onloadend = () => {
        this.setState({
          image: file,
          preview: reader.result,
        });
      };
      reader.readAsDataURL(file);
    }


  /**
   * @description remove add Facility
   *
   * @param {any} event
   *
   * @memberof AddCenter
   *
   * @returns {void}
   */
    removeFacility = (i) => {
      const array = this.state.facility;
      array.splice(i, 1);
      this.setState({ facility: array });
    }

    /**
   * @description renders component to the DOM
   *
   * @memberof AddCenter
   *
   * @returns {JSX} JSX representation of component
   */
    render() {
      const { preview } = this.state;
      let imagePreview = null;
      if (preview) {
        imagePreview = (<img
          src={preview}
          alt="ImagePreview"
          className="imgPre"
        />);
      }

      return (
        <div>
          <div className="container" style={{ paddingTop: '100px' }}>
            <div className="card card w-50 loginCard ">
              <div className="card-header dark">
                <h1 className="color">Add Center</h1>
              </div>
              <h5 style={{ color: 'red', textAlign: 'center', marginTop: '10px' }}>
                { this.state.errorMessage }
              </h5>
              <CenterFrom
                onChange={this.onChange}
                onSubmit={this.onSubmit}
                name={this.state.name}
                city={this.state.city}
                address={this.state.address}
                availability={this.state.availability}
                values={this.state.values}
                handleImageChange={this.handleImageChange}
                onClick={this.onClick}
                removeFacility={this.removeFacility}
                facility={this.state.facility}
                imagePreview={imagePreview}
                about={this.state.about}
                disabled={this.state.values}
                progress={this.state.progress}
                onKeyPress={this.onKeyPress}
                formValid={this.state.formValid}
              />
            </div>
          </div>
        </div>
      );
    }
}

AddCenter.propTypes = {
  initPostCenters: PropTypes.func.isRequired,
  history: PropTypes.shape({}).isRequired,
};

export const mapDispatchToProps = dispatch => ({
  initPostCenters: (input, history) =>
    dispatch(action.initPostCenters(input, history)),
});

export default connect(null, mapDispatchToProps)(AddCenter);
