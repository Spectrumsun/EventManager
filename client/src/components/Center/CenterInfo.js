import React, { Component } from 'react';

import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import * as action from '../../store/actions/index';

class CenterInfo extends Component {
  componentDidMount() {
    this.props.onFetOneCenters(this.props.match.params.id);
  }

  render() {
    const center = (
      <div className="container" style={{ paddingTop: '100px' }}>
        <div className="card loginCard" style={{ width: '45rem' }}>
          <div className="card-header dark">
            <h1 className="color">Center Info</h1>
          </div>
          <div className="card-body">
            <form >
              <div className="form-group row">
                <label htmlFor="staticEmail" className="col-sm-2 col-form-label"><strong>Center name</strong></label>
                <div className="col-sm-10">
                  <input type="text" readOnly className="form-control-plaintext" id="staticEmail" value={this.props.loadedCenter.centerName} />
                </div>
              </div>
              <div className="form-group row">
                <label htmlFor="staticEmail" className="col-sm-2 col-form-label"><strong>Address</strong></label>
                <div className="col-sm-10">
                  <input type="text" readOnly className="form-control-plaintext" id="staticEmail" value={this.props.loadedCenter.address} />
                </div>
              </div>
              <div className="form-group row">
                <label htmlFor="staticEmail" className="col-sm-2 col-form-label"><strong>City</strong></label>
                <div className="col-sm-10">
                  <input type="text" readOnly className="form-control-plaintext" id="staticEmail" value={this.props.loadedCenter.city} />
                </div>
              </div>
              <div className="form-group row">
                <label htmlFor="staticEmail" className="col-sm-2 col-form-label"><strong>Availability</strong></label>
                <div className="col-sm-10">
                  <input type="text" readOnly className="form-control-plaintext" id="staticEmail" value={this.props.loadedCenter.availability} />
                </div>
              </div>
              <h5><strong>Events</strong></h5>
              <ul className="list-group col-md-6">
                {this.props.loadedCenter && this.props.loadedCenter.events && this.props.loadedCenter.events.map(eve =>
                  (<Link to={`/events/${eve.id}`} key={eve.id} style={{ color: 'black' }}>
                    <li className="list-group-item centerlist" key={eve.id} >
                      <strong>{eve.eventName}</strong><h6>{eve.eventdate}</h6>
                    </li>
                   </Link>))}

              </ul>

              <br />
              <h5><strong>Avaliable Facilities</strong></h5>
              <ul className="list-group col-md-4">
                {this.props.loadedCenter && this.props.loadedCenter.facility && this.props.loadedCenter.facility.map(list =>
                    (<li
                      className="list-group-item centerlist"
                      key={this.props.match.params.id}
                    >{list}
                     </li>))}
              </ul>


              <br />
              <button type="submit" className="btn btn-dark" style={{ float: 'left' }}>Edit</button>
            </form>
            <Link to={`${this.props.history.push}/edit`} className="btn btn-danger" style={{ marginLeft: '20px' }}>Delete</Link>
          </div>
        </div>
      </div>
    );


    return (
      <div>
        {center}
      </div>
    );
  }
}


const mapStateToProps = state => ({
  loadedCenter: state.centers.loadedCenter
});

const mapDispatchToProps = dispatch => ({
  onFetOneCenters: id => dispatch(action.getOneCenter(id))
});


export default connect(mapStateToProps, mapDispatchToProps)(CenterInfo);

