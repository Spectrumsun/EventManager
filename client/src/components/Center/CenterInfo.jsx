import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import uuid from 'uuid-random';
import * as action from '../../store/actions/index';
import Footer from '../UI/Footer';
import Loading from '../UI/Loading';
/**
 * @class CenterInfo
 *
 * @extends {React.Component}
 */
export class CenterInfo extends Component {
  /**
   * @description run action on component mount to reload data
   *
   * @param {any} props.params.id
   *
   * @memberof CenterInfo
   */
  componentDidMount() {
    this.props.onFetOneCenters(this.props.match.params.id);
  }

  /**
   * @description deletecCenter a event when the delete button is clicked
   *
   * @param {any} event
   *
   * @memberof CenterInfo
   *
   * @returns {void}
   */
  deleteCenter = (e) => {
    e.preventDefault();
    this.props.onDeleteCenter(
      this.props.match.params.id,
      this.props.history
    );
  }

  /**
   * @description renders component to the DOM
   *
   * @memberof CenterInfo
   *
   * @returns {JSX} JSX representation of component
   */
  render() {
    const admin = this.props.auth.user === null
      ? 'nouser' :
      this.props.auth.user.role;
    const showButton = (
      <div>
        <Link
          to={`/centers/edit/${this.props.match.params.id}`}
          key={this.props.match.params.id}
          style={{ color: '#35434A' }}
        >
          <button
            type="submit"
            id="editCenter"
            className="btn btn-dark"
            style={{ float: 'left' }}
          >Edit
          </button>
        </Link>
        <Link
          to={`${this.props.history.push}/edit`}
          className="btn btn-danger"
          onClick={this.deleteCenter}
          style={{ marginLeft: '20px' }}
          id="deleteCenter"
        >Delete
        </Link>
      </div>
    );
    const { loadedCenter } = this.props;
    const center = (
      <div style={{ paddingTop: '55px', backgroundColor: 'white' }}>
        <div className="card-header dark">
          <h1
            className="container color centerTitle"
          >
            {loadedCenter.centerName}
          </h1>
        </div>
        <img
          className="card-img-top"
          src={loadedCenter.imageurl}
          style={{ height: '600px', width: '100%' }}
          alt="center"
        />
        <div>
          <div
            className="container card-body"
            style={{ width: '45rem' }}
          >
            <h1>
              <strong>{loadedCenter.centerName}</strong>
            </h1>
            <h6 className="list-group-item">{loadedCenter.address}</h6>
            <br />
            <div className="list-group-item centerlist" >
              <h3>About</h3>
              <p className="read" style={{ textAlign: 'justify' }}>
                {loadedCenter.about}
              </p>
            </div>
            <br />
            <h1><strong>City</strong></h1>
            <h6
              className="list-group-item col-md-4"
            >{loadedCenter.city}
            </h6>
            <br />
            <h5 ><strong>Availability</strong></h5>
            <h6
              className="list-group-item col-md-4"
            >{loadedCenter.availability}
            </h6>
            <br />
            <h5><strong>
              Avaliable Facilities
                </strong>
            </h5>
            <ul className="list-group col-md-4 text-capitalize">
              {loadedCenter && loadedCenter.facility && loadedCenter.facility.map(list =>
                (<li
                  className="list-group-item centerlist"
                  key={uuid()}
                >{list}
                 </li>))}
            </ul>
            <br />
            <h5><strong>Events</strong></h5>
            {loadedCenter && loadedCenter.events && loadedCenter.events.map(eventDate =>
                  (<div
                    key={Math.floor(Math.random() * 10906) + 30}
                    className="card d-lg-inline-block"
                    style={{ width: '15rem' }}
                  >
                    <div className="card-body">
                      <h6 className="card-text">
                        {eventDate.startDate} To {eventDate.endDate}
                      </h6>
                    </div>
                   </div>
                ))}
            <br />
            {admin === process.env.VALUE ? showButton : null}
          </div>
        </div>
        <Footer />
      </div>
    );

    return (
      <div>
        {
          this.props.loadedCenter === undefined ||
          this.props.error !== false ? <Loading /> : center
        }
      </div>
    );
  }
}

CenterInfo.propTypes = {
  onDeleteCenter: PropTypes.func.isRequired,
  onFetOneCenters: PropTypes.func.isRequired,
  error: PropTypes.bool,
  auth: PropTypes.shape({
    isAuthenticated: PropTypes.bool,
    user: PropTypes.shape({
      id: PropTypes.number,
      role: PropTypes.string,
    })
  }).isRequired,
  history: PropTypes.shape({
    push: PropTypes.func,
  }).isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string,
    }),
  }).isRequired,

  loadedCenter: PropTypes.shape({
    id: PropTypes.number,
    eventName: PropTypes.string,
    startDate: PropTypes.string,
    endDate: PropTypes.string,
    time: PropTypes.string,
    purpose: PropTypes.string,
  }),
};

CenterInfo.defaultProps = {
  loadedCenter: PropTypes.shape({
    userId: 1,
    centerName: 'eventName',
    address: '2018-10-02',
    time: '11:00',
    purpose: 'fun'
  }),
  error: false
};


const mapStateToProps = state => ({
  loadedCenter: state.centers.loadedCenter,
  auth: state.auth,
  error: state.centers.error
});

const mapDispatchToProps = dispatch => ({
  onFetOneCenters: id =>
    dispatch(action.getOneCenter(id)),
  onDeleteCenter: (id, history) =>
    dispatch(action.initDeleteCenter(id, history))
});


export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CenterInfo);
