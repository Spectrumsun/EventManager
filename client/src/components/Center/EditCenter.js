import React, { Component } from 'react';
import toast from 'toastr';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import CenterFrom from './Form/CenterForm';
import * as action from '../../store/actions/index';

class AddCenter extends Component {
    state = {
      name: this.props.loadedCenter.centerName,
      city: this.props.loadedCenter.city,
      address: this.props.loadedCenter.address,
      availability: this.props.loadedCenter.availability,
      facility: this.props.loadedCenter.facility || [],
      values: '',
      image: this.props.loadedCenter.imageurl,
      publicid: this.props.loadedCenter.imageId,
    }

    onClick = () => {
      this.setState({ facility: this.state.facility.concat([this.state.values]) });
      this.setState({ values: '' });
    }


  onChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  }

   onSubmit = (e) => {
     if (e.target.type != 'textarea' && e.which === 13 /* Enter */) {
       e.preventDefault();
     }
     e.preventDefault();
     if (this.state.name === '') {
       toast.error('Center Name cannot be blank');
     } else if (this.state.date === '') {
       toast.error('Center city cannot be blank');
     } else if (this.state.time === '') {
       toast.error('Center Address cannot be blank');
     } else if (this.state.purpose === '') {
       toast.error('Center Availability must be set');
     } else {
       this.props.initEditCenter(
         this.props.match.params.id,
         this.state, this.props.history
       );
     }
   }

    removeFacility = (i) => {
      const array = this.state.facility;
      array.splice(i, 1);
      this.setState({ facility: array });
    }


    render() {
      return (
        <div className="container" style={{ paddingTop: '100px' }}>
          <div className="card card w-50 loginCard ">
            <div className="card-header dark">
              <h1 className="color">Edit Center</h1>
            </div>
            <CenterFrom
              onChange={this.onChange}
              onSubmit={this.onSubmit}
              name={this.state.name}
              city={this.state.city}
              address={this.state.address}
              availability={this.state.availability}
              values={this.state.values}
              image={this.state.image}
              publicid={this.state.publicid}
              onClick={this.onClick}
              facility={this.state.facility}
              removeFacility={this.removeFacility}
              disabled={this.state.values}
            />
          </div>
        </div>
      );
    }
}

AddCenter.propTypes = {
  initEditCenter: PropTypes.func.isRequired,
  loadedCenter: PropTypes.shape({
    centerName: PropTypes.string.isRequired,
    city: PropTypes.string.isRequired,
    address: PropTypes.string.isRequired,
    availability: PropTypes.string.isRequired,
    facility: PropTypes.array.isRequired,
    imageurl: PropTypes.string.isRequired,
    imageId: PropTypes.string.isRequired,
  }),
  history: PropTypes.shape({}).isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string,
    }),
  }).isRequired,
};


AddCenter.defaultProps = {
  loadedCenter: PropTypes.shape({
    userId: 1,
    eventName: 'eventName',
    eventdate: '2018-10-02',
    time: '11:00',
    purpose: 'fun',
    facility: ['example']
  }),
};


const mapStateToProps = state => ({
  loadedCenter: state.centers.loadedCenter,
});

const mapDispatchToProps = dispatch => ({
  initEditCenter: (id, input, history) => dispatch(action.initEditCenter(id, input, history)),
});


export default connect(mapStateToProps, mapDispatchToProps)(AddCenter);
