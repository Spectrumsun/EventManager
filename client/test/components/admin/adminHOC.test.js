import React from 'react';
import { shallow } from 'enzyme';
import IsAdmin from '../../../src/components/Auth/adminHOC';
import { AddCenter } from '../../../src/components/Center/AddCenter'

const components = IsAdmin(AddCenter);

const { WrappedComponent } = components;


let props;
let mountedComponent;

/**
 * @description Initialise the component
 *
 * @returns {object} ManageRecipe - Mounted component
 */
const getComponent = () => {
  if (!mountedComponent) {
    mountedComponent = shallow(<WrappedComponent {...props} />);
  }
  return mountedComponent;
};

/**
 * @description Initialise the component
 *
 * @returns {object} ManageRecipe - Mounted component
 */

describe('Component: IsAdminenticate Admin', () => {
  beforeEach(() => {
    props = {
      isAuthenticated: true,
      auth: {
        isAuthenticated: true,
        user: {
          id: 1,
          role: 'user'
        }
      },
      history: {
        push: jest.fn()
      }
    };
    mountedComponent = undefined;
  });

  describe('Protected Routes test suite', () => {
    it('Redirects user to homepage they are not Admin', () => {
      props.isAuthenticated = false;
      expect(getComponent()).toMatchSnapshot();
    });
  });
});

