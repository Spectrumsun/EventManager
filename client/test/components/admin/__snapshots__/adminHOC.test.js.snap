// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: IsAdminenticate Admin Protected Routes test suite Redirects user to homepage they are not Admin 1`] = `
<AddCenter
  auth={
    Object {
      "isAuthenticated": true,
      "user": Object {
        "id": 1,
        "role": "user",
      },
    }
  }
  history={
    Object {
      "push": [MockFunction] {
        "calls": Array [
          Array [
            "/",
          ],
        ],
      },
    }
  }
  isAuthenticated={false}
/>
`;
